tests/helpers/guildSettingsPage.js

```
@@ -66,16 +69,22 @@ function bootPage({ panelFetch } = {}) {
+    const addWindowListener = window.addEventListener.bind(window);
     document.addEventListener = (type, fn, opts) => {
-        documentListeners.push([type, fn, opts]);
+        pageListeners.push([document, type, fn, opts]);
         addDocumentListener(type, fn, opts);
     };
+    window.addEventListener = (type, fn, opts) => {
+        pageListeners.push([window, type, fn, opts]);
+        addWindowListener(type, fn, opts);
+    };
     try {
         window.eval(fs.readFileSync(path.join(PUBLIC, 'esc-html.js'), 'utf8'));
```

Biome (2.5.7)

[error] eval() exposes to security risks and performance issues.

(lint/security/noGlobalEval)